src/main/java/de/adesso/service/GitRepoDiffer.java

```diff
@@ -52,48 +46,44 @@ public GitRepoDiffer(ConfigService configService) {
      * command. Changed files will be logged
      */
     public List<DiffEntry> checkForUpdates() {
-        LOGGER.info("Checking for Updates");
+        LOGGER.info("Checking for updates...");
         try {
-            repo = LocalRepoCreater.getLocalGit().getRepository();
+            Repository repo = LocalRepoCreater.getLocalGit().getRepository();
             ObjectReader reader = repo.newObjectReader();
 
-            // get current head (latest commit) from remote
-            CanonicalTreeParser newTreeIter = new CanonicalTreeParser();
-            ObjectId newTree = repo.resolve("HEAD^{tree}");
-            newTreeIter.reset(reader, newTree);
-
             // search latest commit with changes done by a contributor who is not GIT_AUTHOR_NAME
             // this commit is the last commit which is not done by GIT_AUTHOR_NAME
-            LOGGER.info("Searching latest commit not done by " + configService.getGIT_AUTHOR_NAME());
-            CanonicalTreeParser oldTreeIter = new CanonicalTreeParser();
-            RevCommit revCommitOld = StreamSupport.stream(new Git(repo).log().all().call().spliterator(), false)
-                    .filter(commit -> !commit.getAuthorIdent().getName().equals(configService.getGIT_AUTHOR_NAME()))
+            LOGGER.info("Searching latest commit not done by " + configService.getGIT_AUTHOR_NAME() + "...");
+            CanonicalTreeParser newTreeIter = new CanonicalTreeParser();
+            RevCommit revCommitNew = StreamSupport.stream(new Git(repo).log().all().call().spliterator(), false)
+                    .filter(commit -> !commit.getAuthorIdent().getName().equals(configService.getGIT_AUTHOR_NAME()) && checkOnPost(commit))
                     .findFirst().orElse(null);
```

Comment:
`search latest commit with changes done by a contributor who is not GIT_AUTHOR_NAME`
Klingt ganz so, also hättet ihr hier gerade den neuen Methodennamen des kommenden Blockes beschrieben :)

Dementsprechend: Den Block in eine eigene Methode auslagern.